### PR TITLE
Use nameof for some literals

### PR DESCRIPTION
**Pull request by blairconrad**

### CODE REVIEW

File: src/FakeItEasy/Core/DelegateRaiser.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ private static void AssertThatValuesSatisfyCallSignature(MethodInfo callMethod,
 
             throw new FakeConfigurationException(
                 "The event has the signature ({0}), but the provided arguments have types ({1})."
-                    .FormatInvariant(fakeSignature, actionSignature, "Raise"));
+                    .FormatInvariant(fakeSignature, actionSignature, nameof(Raise)));
```
Comment (blairconrad) [inline src/FakeItEasy/Core/DelegateRaiser.cs]: In retrospect, this one may be an error. It refers to the class `Raise`. I will restore.

File: src/FakeItEasy/Creation/DelegateProxies/DelegateProxyGenerator.cs [hunk 2]
```diff
@@ -163,7 +163,7 @@ private static NewArrayExpression WrapParametersInObjectArray(ParameterExpressio
 
         private class DelegateCallInterceptedEventRaiser
         {
-            public static readonly MethodInfo RaiseMethod = typeof(DelegateCallInterceptedEventRaiser).GetMethod("Raise");
+            public static readonly MethodInfo RaiseMethod = typeof(DelegateCallInterceptedEventRaiser).GetMethod(nameof(Raise));
```
Comment (blairconrad) [inline src/FakeItEasy/Creation/DelegateProxies/DelegateProxyGenerator.cs]: Mostly I wanted the change so I'd stop getting complaints that `Raise` wasn't used.

**Comment by blairconrad** [comment]
Oh, thanks for the label! So forgetful. Sigh.

File: src/FakeItEasy/Core/DelegateRaiser.cs (hunk 1 above)
Comment (thomaslevesque) [inline src/FakeItEasy/Core/DelegateRaiser.cs]: Actually, I think `"Raise"` could be removed... there are only 2 placeholders in the format strings, not 3

File: src/FakeItEasy/Core/DelegateRaiser.cs [hunk 3]
```diff
@@ -5,7 +5,7 @@ namespace FakeItEasy.Core
     using System.Diagnostics.CodeAnalysis;
     using System.Linq;
     using System.Reflection;
-
+    using Castle.DynamicProxy.Generators;
```
Comment (thomaslevesque) [inline src/FakeItEasy/Core/DelegateRaiser.cs]: Is this intentional?

Comment (blairconrad) [inline src/FakeItEasy/Core/DelegateRaiser.cs]: Ah. No. I have no explanation.

Comment (blairconrad) [inline src/FakeItEasy/Core/DelegateRaiser.cs]: Ah. Not true. I think it came from the change to `nameof(Raise)` that I incompletely reverted.

File: src/FakeItEasy/Core/DelegateRaiser.cs (hunk 1 above)
Comment (blairconrad) [inline src/FakeItEasy/Core/DelegateRaiser.cs]: YES!

**Comment by blairconrad** [comment]
Thanks for the keen reviews, @thomaslevesque.

**Comment by thomaslevesque** [comment]
LGTM :+1:

I'll merge when the build completes

File: src/FakeItEasy/Creation/DelegateProxies/DelegateProxyGenerator.cs (hunk 2 above)
Comment (adamralph) [inline src/FakeItEasy/Creation/DelegateProxies/DelegateProxyGenerator.cs]: Does this still refer to the "Raise" type rather than the method?

**Comment by adamralph** [comment]
Please see https://github.com/FakeItEasy/FakeItEasy/pull/792#discussion_r69491028 before merging.

File: src/FakeItEasy/Creation/DelegateProxies/DelegateProxyGenerator.cs (hunk 2 above)
Comment (thomaslevesque) [inline src/FakeItEasy/Creation/DelegateProxies/DelegateProxyGenerator.cs]: No, it refers to the method, which is closer in scope (just checked). If we want to be extra careful, we could fully qualify it (`DelegateCallInterceptedEventRaiser.Raise`).

Comment (adamralph) [inline src/FakeItEasy/Creation/DelegateProxies/DelegateProxyGenerator.cs]: OK, I think it's fine as is.

**Pull request merged**